packages/pigeon/lib/ast.dart

```diff
@@ -82,6 +74,30 @@ class Api extends Node {
   }
 }
 
+/// A parameter to a generic entity.
+class TypeArgument {
+  /// Constructor for [TypeArgument].
+  TypeArgument({
+    required this.dataType,
+    required this.isNullable,
+    this.typeArguments,
+  });
+
+  /// A string representation of the base datatype.
+  final String dataType;
+
+  /// The type arguments to [TypeArgument].
+  final List<TypeArgument>? typeArguments;
+
+  /// True if the type is nullable.
+  final bool isNullable;
+
+  @override
+  String toString() {
+    return '(TypeArgument dataType:$dataType isNullable:$isNullable typeArguments:$typeArguments)';
```

Comment:
Wouldn't it be more useful to turn this into the type string people are used to dealing with? I.e., so that it would print:
```
(TypeArgument: List<String?>)
```

rather than:

```
(TypeArgument dataType:List isNullable:false typeArguments:[(TypeArgument dataType:String isNullable:true typeArguments:null)])
```

Reply:
Yea, that's a fair bit more concise but it doesn't match the pattern I'm using elsewhere.  It's just easier and more robust to follow the format and be verbose without thinking about it imo.

Reply:
The point of toString is for human consumption; I'm having a hard time imagining a case where I'd want to see the second instead of the first while debugging.

What is the use case where the the second form is easier to read than the first?

Reply:
It tells you more in the the same way a debugger does by just telling you the raw data.  A simple sexp parser could parse the output of toString to create a recreation of the data.  Any logic to make it more concise might demonstrate the same bug someone is trying to debug.